File: WinFormHelper.cs
int x, y;
Point newpoint = new Point();

#region FormMove

        private void frmMain_MouseDown(object sender, MouseEventArgs e)
        {
            x = Control.MousePosition.X - this.Location.X;
            y = Control.MousePosition.Y - this.Location.Y;
        }

        

        private void frmMain_MouseMove(object sender, MouseEventArgs e)
        {
            if (e.Button == MouseButtons.Left)
            {
                newpoint = Control.MousePosition;
                newpoint.X -= x;
                newpoint.Y -= y;
                this.Location = newpoint;
                Application.DoEvents();
            }
        }
        #endregion

File: WinFormHelper.vb
'--FORM MOVE--'

Dim x, y As Integer
Dim newpoint As New Point

#Region "Form Move"
    Private Sub frmMain_MouseDown(sender As Object, e As MouseEventArgs) Handles Me.MouseDown, panelTitle.MouseDown, labelTitle.MouseDown
        x = Control.MousePosition.X - Me.Location.X
        y = Control.MousePosition.Y - Me.Location.Y
    End Sub



    Private Sub frmMain_MouseMove(sender As Object, e As MouseEventArgs) Handles Me.MouseMove, panelTitle.MouseMove, labelTitle.MouseMove
        If e.Button = MouseButtons.Left Then
            newpoint = Control.MousePosition
            newpoint.X -= x
            newpoint.Y -= y
            Me.Location = newpoint
            Application.DoEvents()
        End If
    End Sub


#End Region

'--END FORM MOVE--'